source/extensions/watchdog/abort_action/abort_action.cc

```diff
@@ -0,0 +1,69 @@
+#include "extensions/watchdog/abort_action/abort_action.h"
+
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+
+#ifndef WIN32
+#include <sys/types.h>
+#include <csignal>
+#endif
+
+#include "common/common/logger.h"
+#include "common/protobuf/utility.h"
+#include "envoy/thread/thread.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Watchdog {
+namespace AbortAction {
+namespace {
+#ifdef __linux__
+pid_t toPlatformTid(int64_t tid) { return static_cast<pid_t>(tid); }
+#elif defined(__APPLE__)
+uint64_t toPlatformTid(int64_t tid) { return static_cast<uint64_t>(tid); }
+#endif
+} // namespace
+
+AbortAction::AbortAction(
+    envoy::extensions::watchdog::abort_action::v3alpha::AbortActionConfig& config,
+    Server::Configuration::GuardDogActionFactoryContext& /*context*/)
+    : config_(config){};
+
+void AbortAction::run(
+    envoy::config::bootstrap::v3::Watchdog::WatchdogAction::WatchdogEvent /*event*/,
+    const std::vector<std::pair<Thread::ThreadId, MonotonicTime>>& thread_last_checkin_pairs,
+    MonotonicTime /*now*/) {
+
+  if (thread_last_checkin_pairs.empty()) {
+    ENVOY_LOG_MISC(warn, "Watchdog AbortAction called without any thread.");
+    return;
+  }
+
+  int64_t raw_tid = thread_last_checkin_pairs[0].first.getId();
+
+#ifdef WIN32
+  // TODO(kbaichoo): add support for this with windows.
+  ENVOY_LOG_MISC(error, "Watchdog AbortAction is unimplemented for Windows.");
```

Comment:
Is this actually needed? I think Windows has its own extension bzl file so I'm not sure this is even compiled? Could we actually avoid the ifdef in here now or just replace with an proprocessor error if this extension gets compiled on windows?

Reply:
Good to know this is a possibility, thanks for pointing it out.

I've added the extension to `WINDOWS_SKIP_TARGETS` which IIUC will let it be skipped by windows so I can remove the `ifdef #Win32` from code.

Reply:
turns out you also need to add `tags = ["skip_on_windows"]` on tests since excluding the extension in the `.bzl` doesn't effect the tests.